components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -49,23 +78,34 @@ internal class RustPushConnection(
     internal var api: PushAPI? = null
 
     @GuardedBy("this")
-    override suspend fun subscribe(channelId: String, scope: String): SubscriptionResponse = synchronized(this) {
+    override suspend fun subscribe(
+        scope: PushScope,
+        appServerKey: AppServerKey?
+    ): AutoPushSubscription = synchronized(this) {
         val pushApi = api
         check(pushApi != null) { "Rust API is not initiated; updateToken hasn't been called yet." }
-        return pushApi.subscribe(channelId, scope)
+
+        val channelId = scope.toChannelId()
```

Comment:
Somewhere here would be an excellent place to describe in a comment that Autopush really operates over these `channelID` things, while the WebPush world is built around a similar (but different) concept of `scope`. Here is the point of transition.

Reply:
I've added implementation notes for the `RustPushConnection` to document the relation between those two concepts.